Tests/JobsPostgresTests/JobsTests.swift

```diff
@@ -683,4 +683,121 @@ final class JobsTests: XCTestCase {
             group.cancelAll()
         }
     }
+
+    func testResumableAndPausableJobs() async throws {
+        struct TestParameters: JobParameters {
+            static let jobName = "testResumableAndPausableJobs"
+            let value: Int
+        }
+        let expectation = XCTestExpectation(description: "TestJob.execute was called", expectedFulfillmentCount: 2)
+        let jobExecutionSequence: NIOLockedValueBox<[Int]> = .init([])
+
+        let jobQueue = try await self.createJobQueue(numWorkers: 1, configuration: .init(), function: #function)
+
+        try await testPriorityJobQueue(jobQueue: jobQueue) { queue in
+            queue.registerJob(parameters: TestParameters.self) { parameters, context in
+                context.logger.info("Parameters=\(parameters.value)")
+                jobExecutionSequence.withLockedValue {
+                    $0.append(parameters.value)
+                }
+                try await Task.sleep(for: .milliseconds(Int.random(in: 10..<50)))
+                expectation.fulfill()
+            }
+
+            let firstJob = try await queue.push(
+                TestParameters(value: 20),
+                options: .init(
+                    priority: .lowest()
+                )
+            )
+
+            try await jobQueue.pauseJob(jobID: firstJob)
+
+            try await queue.push(
+                TestParameters(value: 2025),
+                options: .init(
+                    priority: .normal()
+                )
+            )
+
+            try await withThrowingTaskGroup(of: Void.self) { group in
+                let serviceGroup = ServiceGroup(services: [queue], logger: queue.logger)
+
+                let processingJobs = try await jobQueue.queue.getJobs(withStatus: .pending)
+                XCTAssertEqual(processingJobs.count, 1)
+
+                group.addTask {
+                    try await serviceGroup.run()
+                }
+
+                let pausedJobs = try await jobQueue.queue.getJobs(withStatus: .paused)
+                XCTAssertEqual(pausedJobs.count, 1)
+
+                try await jobQueue.resumeJob(jobID: firstJob)
```

Comment:
This test is testing that resumed jobs get placed at the end of the queue.
Perhaps we need another test that does the following
- Add job1
- Add job2
- Pause job1
- Start job queue
- Wait for job2
- Verify job1 didn't run
- Resume job1
- Wait for job1